### PR TITLE
vlb loadbalancer info module

### PR DESCRIPTION
**Pull request by Polina-Gubina**

### CODE REVIEW

File: meta/runtime.yml [hunk 1]
```diff
@@ -63,6 +63,7 @@ action_groups:
     - volume_backup
     - waf_certificate
     - waf_certificate_info
+    - vlb_loadbalancer_info
```
Comment (anton-sidelnikov) [inline meta/runtime.yml:66]: lets call it dlb_loadbalancer_info (as dedicated load balancer)

File: plugins/modules/vlb_loadbalancer_info.py [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+  vip_port_id:
+    description:
+      - Specifies the ID of the port bound to the virtual IP address of the
+        load balancer.
+    type: str
+  vip_address:
+    description:
+      - Specifies the virtual IP address bound to the load balancer.
+    type: str
+  vip_subnet_cidr_id:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:56]: subnet_id

File: plugins/modules/vlb_loadbalancer_info.py [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+  vpc_id:
+    description:
+      - Specifies the ID of the VPC where the load balancer works.
+    type: str
+  vip_port_id:
+    description:
+      - Specifies the ID of the port bound to the virtual IP address of the
+        load balancer.
+    type: str
+  vip_address:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:52]: ip_address

File: plugins/modules/vlb_loadbalancer_info.py [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+    description:
+      - Specifies the operating status of the load balancer.
+        The value can only be ONLINE, indicating that the load balancer
+        is working normally.
+    type: str
+  vpc_id:
+    description:
+      - Specifies the ID of the VPC where the load balancer works.
+    type: str
+  vip_port_id:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:47]: port_id

File: plugins/modules/vlb_loadbalancer_info.py [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+    type: str
+  l4_flavor_id:
+    description:
+      - Specifies the ID of the flavor at Layer 4.
+    type: str
+  l4_scale_flavor_id:
+    description:
+      - Specifies the elastic flavor that is reserved for now.
+    type: str
+  availability_zone_list:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:68]: availability_zones

File: plugins/modules/vlb_loadbalancer_info.py [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+    elements: str
+  l7_flavor_id:
+    description:
+      - Specifies the ID of the flavor at Layer 7.
+    type: str
+  l7_scale_flavor_id:
+    description:
+      - Specifies the elastic flavor that is reserved for now.
+    type: str
+  member_device_id:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:86]: member_id

File: plugins/modules/vlb_loadbalancer_info.py [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+  publicips:
+    description:
+      - Specifies the public IP address bound to the load balancer.
+    type: list
+    elements: str
+  ip_version:
+    description:
+      - Specifies the IP version. The value can be only 4 (IPv4).
+    type: int
+  elb_virsubnet_type:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:105]: subnet_type

File: plugins/modules/vlb_loadbalancer_info.py [hunk 8]
```diff
@@ -0,0 +1,386 @@
...
+    description:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:106]: check description: loadbalancers -> contains -> elb_virsubnet_type -> description -> 2\n  str type expected (type=type_error.str)']:plugin_name=opentelekomcloud.cloud.vlb_loadbalancer_info:plugin_type=module|opentelekomcloud.cloud.vlb_loadbalancer_info did not return correct RETURN or EXAMPLES

File: plugins/modules/vlb_loadbalancer_info.py [hunk 9]
```diff
@@ -0,0 +1,386 @@
...
+          type: str
+        publicip_address:
+          description: Specifies the IP address.
+          type: str
+        ip_version:
+          description:
+            - Specifies the IP version. The value can be 4 (IPv4) or 6 (IPv6).
+          type: int
+    elb_virsubnet_ids:
+      description:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:282]: try to rename this field also in response to: network_ids

File: plugins/modules/vlb_loadbalancer_info.py [hunk 10]
```diff
@@ -0,0 +1,386 @@
...
+        - Specifies the ID of the subnet on the downstream plane.
+        - The ports used by the load balancer dynamically occupy IP addresses.
+          in the subnet.
+      type: list
+    elb_virsubnet_type:
+      description:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:288]: and here: loadbalancers -> contains -> elb_virsubnet_type -> description -> 2\n  str type expected (type=type_error.str)']:plugin_name=opentelekomcloud.cloud.vlb_loadbalancer_info:plugin_type=module|opentelekomcloud.cloud.vlb_loadbalancer_info did not return correct RETURN or EXAMPLES

File: plugins/modules/vlb_loadbalancer_info.py [hunk 11]
```diff
@@ -0,0 +1,386 @@
...
+          description:
+            - Specifies the IP version. The value can be 4 (IPv4) or 6 (IPv6).
+          type: int
+    elb_virsubnet_ids:
+      description:
+        - Specifies the ID of the subnet on the downstream plane.
+        - The ports used by the load balancer dynamically occupy IP addresses.
+          in the subnet.
+      type: list
+    elb_virsubnet_type:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:287]: subnet_type

File: plugins/modules/vlb_loadbalancer_info.py [hunk 12]
```diff
@@ -0,0 +1,386 @@
...
+        - Specifies the private IPv4 address bound to the load balancer.
+      type: str
+    vip_subnet_cidr_id:
+      description:
+        - Specifies the ID of the IPv4 subnet where the load balancer works.
+      type: str
+    project_id:
+      description: Specifies the project ID of the load balancer.
+      type: str
+    vip_port_id:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:195]: try to rename this field also in response to: port_id

File: plugins/modules/vlb_loadbalancer_info.py [hunk 13]
```diff
@@ -0,0 +1,386 @@
...
+    operating_status:
+      description:
+        - Specifies the operating status of the load balancer.
+          The value can only be ONLINE.
+      type: str
+    vip_address:
+      description:
+        - Specifies the private IPv4 address bound to the load balancer.
+      type: str
+    vip_subnet_cidr_id:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:188]: try to rename this field also in response to: subnet_id

File: plugins/modules/vlb_loadbalancer_info.py [hunk 14]
```diff
@@ -0,0 +1,386 @@
...
+      contains:
+        id:
+          description: ID of the listener.
+          type: str
+    operating_status:
+      description:
+        - Specifies the operating status of the load balancer.
+          The value can only be ONLINE.
+      type: str
+    vip_address:
```
Comment (anton-sidelnikov) [inline plugins/modules/vlb_loadbalancer_info.py:184]: try to rename this field also in response to: ip_address

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: meta/runtime.yml (hunk 1 above)
Comment (anton-sidelnikov) [reply meta/runtime.yml:66]: or better even call it simply loadbalancer_info, because it will be unified for both type of loadbalancers

**Comment by anton-sidelnikov** [comment]
https://github.com/opentelekomcloud/ansible-collection-cloud/blob/main/tests/sanity/ignore.txt also add your module here

**Review by artem-lifshits — DISMISSED** [review]

File: plugins/modules/loadbalancer_v3_info.py [hunk 15]
```diff
@@ -0,0 +1,400 @@
...
+  publicips:
+    description:
+      - Specifies the public IP address bound to the load balancer.
+    type: list
+    elements: str
+  ip_version:
+    description:
+      - Specifies the IP version. The value can be only 4 (IPv4).
+    type: int
+  elb_virsubnet_type:
```
Comment (anton-sidelnikov) [inline plugins/modules/loadbalancer_v3_info.py:104]: subnet_type

File: plugins/modules/loadbalancer_v3_info.py [hunk 16]
```diff
@@ -0,0 +1,400 @@
...
+          description: Specifies the EIP ID.
+          type: str
+        publicip_address:
+          description: Specifies the IP address.
+          type: str
+        ip_version:
+          description:
+            - Specifies the IP version. The value can be 4 (IPv4) or 6 (IPv6).
+          type: int
+    elb_virsubnet_ids:
```
Comment (anton-sidelnikov) [inline plugins/modules/loadbalancer_v3_info.py:280]: subnet_ids

File: plugins/modules/loadbalancer_v3_info.py [hunk 17]
```diff
@@ -0,0 +1,400 @@
...
+      description:
+        - Specifies the ID of the subnet on the downstream plane.
+        - The ports used by the load balancer dynamically occupy IP addresses.
+          in the subnet.
+      type: list
+    elb_virsubnet_type:
```
Comment (anton-sidelnikov) [inline plugins/modules/loadbalancer_v3_info.py:286]: subnet_type

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by artem-lifshits — APPROVED** [review]

**Pull request merged**